internal/plan/plan.go

```diff
@@ -55,6 +56,79 @@ type Layer struct {
 	Description string              `yaml:"description,omitempty"`
 	Services    map[string]*Service `yaml:"services,omitempty"`
 	Checks      map[string]*Check   `yaml:"checks,omitempty"`
+	Logging     *Logging            `yaml:"logging,omitempty"`
+}
+
+type Logging struct {
+	Labels       map[string]string          `yaml:"labels,omitempty"`
+	Destinations map[string]*LogDestination `yaml:"destinations,omitempty"`
+}
+
+func (l *Logging) Copy() *Logging {
+	copied := *l
+	copied.Labels = make(map[string]string)
+	for k, v := range l.Labels {
+		copied.Labels[k] = v
+	}
+	copied.Destinations = make(map[string]*LogDestination)
+	for k, v := range l.Labels {
+		copied.Labels[k] = v
```

Comment:
Yeah, this is the problem with hand-writing these that I've run into before. :-) This should be `copied.Destinations[k] = v`.